Does not work on true no_std, bare metal target

This is compiling for a amd64 kernel that does not have libc yet.

```
   Compiling yakern v0.1.0 (C:\Users\steve\rust\yakern\yakern)
   Compiling safer-ffi v0.0.10
error[E0432]: unresolved import `libc::size_t`                                                                                                                                                                                                                                                                         
  --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\utils\prelude.rs:29:19
   |
29 | pub(in crate) use ::libc::size_t;
   |                   ^^^^^^^^^^^^^^ no `size_t` in the root

error[E0412]: cannot find type `uintptr_t` in crate `libc`                                                                                                                                                                                                                                                             
 --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\layout\impls.rs:4:36
  |
4 |     ::core::mem::size_of::<::libc::uintptr_t>()
  |                                    ^^^^^^^^^ not found in `libc`

error[E0412]: cannot find type `size_t` in crate `libc`
 --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\layout\impls.rs:6:36
  |
6 |     ::core::mem::size_of::<::libc::size_t>()
  |                                    ^^^^^^ not found in `libc`

error[E0412]: cannot find type `c_char` in crate `libc`
  --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\c_char.rs:35:39
   |
35 |         let _ = is_u8_or_i8::<::libc::c_char>;
   |                                       ^^^^^^ not found in `libc`
   |
help: consider importing one of these items
   |
1  | use core::ffi::c_char;
   |
1  | use crate::c_char;
   |
1  | use crate::core::ffi::c_char;
   |
help: if you import `c_char`, refer to it directly
   |
35 -         let _ = is_u8_or_i8::<::libc::c_char>;
35 +         let _ = is_u8_or_i8::<c_char>;
   |

error[E0080]: evaluation of constant value failed                                                                                                                                                                                                                                                                      
   --> C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\utils\macros.rs:105:13
    |
105 |               COND
    |               ^^^^ referenced constant has errors
    |
   ::: C:\Users\steve\.cargo\registry\src\github.com-1ecc6299db9ec823\safer-ffi-0.0.10\src\layout\impls.rs:3:1
    |
3   | / const_assert! {
4   | |     ::core::mem::size_of::<::libc::uintptr_t>()
5   | |     ==
6   | |     ::core::mem::size_of::<::libc::size_t>()
7   | | }
    | |_- in this macro invocation
    |
    = note: this error originates in the macro `const_assert` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -60,6 +60,7 @@ log = { version = "0.4.8", optional = true }
 require_unsafe_in_body = { version = "0.2.1", optional = true }
 inventory = { version = "0.1.6", optional = true }
 proc-macro-hack = { version = "0.5.15", optional = true }
+cfg-if = "1.0.0"
 
 [dependencies.safer_ffi-proc_macro]
 path = "src/proc_macro"

--- a/src/c_char.rs
+++ b/src/c_char.rs
@@ -16,7 +16,7 @@ struct c_char /* = */ (
 );
 
 /// Assert that `::libc::c_char` is either `uint8_t` or `int8_t`.
-#[cfg(not(target_arch = "wasm32"))] // no libc on WASM
+#[cfg(not(any(target_arch = "wasm32", not(feature = "std"))))] // no libc on WASM nor no_std
 const _: () = {
     trait IsU8OrI8
     {}

--- a/src/layout/impls.rs
+++ b/src/layout/impls.rs
@@ -1,5 +1,6 @@
 use super::*;
 
+#[cfg(not(any(target_arch = "wasm32", not(feature = "std"))))] // no libc on WASM nor no_std
 const_assert! {
     ::core::mem::size_of::<::libc::uintptr_t>()
     ==

--- a/src/utils/prelude.rs
+++ b/src/utils/prelude.rs
@@ -25,12 +25,18 @@ pub(in crate) use ::core::{
         Not as _,
     },
 };
-#[cfg(not(target_arch = "wasm32"))]
-pub(in crate) use ::libc::size_t;
 
-#[cfg(target_arch = "wasm32")]
-#[allow(bad_style)]
-pub(in crate) type size_t = u32;
+cfg_if::cfg_if! {
+    if #[cfg(not(feature = "std"))] {
+        #[allow(bad_style)]
+        pub(in crate) type size_t = usize;
+    } else if #[cfg(target_arch = "wasm32")] {
+        #[allow(bad_style)]
+        pub(in crate) type size_t = u32;
+    } else {
+        pub(in crate) use ::libc::size_t;
+    }
+}
 
 cfg_alloc! {
     pub(in crate) use ::alloc::{

--- a/src/utils/prelude.rs
+++ b/src/utils/prelude.rs
@@ -27,12 +27,12 @@ pub(in crate) use ::core::{
 };
 
 cfg_if::cfg_if! {
-    if #[cfg(not(feature = "std"))] {
+    if #[cfg(any(
+        not(feature = "std"),
+        target_arch = "wasm32",
+    ))] {
         #[allow(bad_style)]
         pub(in crate) type size_t = usize;
-    } else if #[cfg(target_arch = "wasm32")] {
-        #[allow(bad_style)]
-        pub(in crate) type size_t = u32;
     } else {
         pub(in crate) use ::libc::size_t;
     }

